addon/services/activity-tracking.ts

```diff
@@ -0,0 +1,72 @@
+import Service, { inject as service } from '@ember/service';
+import Configuration from '@upfluence/hyperevents/configuration';
+
+export type ActivityType = 'page_view' | 'button_click' | 'component_view';
+export type Activity = {
+  type: ActivityType;
+  action?: string;
+  path: string;
+  route: string;
+  engine?: string;
+  version?: string;
+  extra?: {};
+};
+
+const RETRY_ATTEMPTS: number = 1;
+
+export default class ActivityTracking extends Service {
+  @service declare session: any;
+
+  log(activity: Activity): void {
+    // this method could also batch all the activities during XX seconds
+    // and then make the call to the BE
+    // Downside of doing this is that we might lose information if the user closes the page/app
+    this.performCall([activity]);
+  }
+
+  logMultiple(activities: Activity[]): void {
+    this.performCall(activities);
+  }
+
+  private performCall(activities: Activity[], tries: number = RETRY_ATTEMPTS): void {
+    this.sendBulkActivities(activities)
+      .then(() => {
+        console.log('Activities have been logged');
+        console.log(activities);
+      })
+      .catch(() => {
+        if (tries > 0) this.performCall(activities, --tries);
+      });
+  }
+
+  private sendBulkActivities(activities: Activity[]): Promise<void> {
+    return fetch(this.ApiUrl, {
+      method: 'POST',
+      headers: this.headers,
+      body: JSON.stringify(activities)
+    }).then((response: Response) => {
+      if (!response.ok) {
+        return response.json().then((e) => Promise.reject(e));
```

Comment:
The backend send us a paid for errors, right?

Reply:
AFAIK, the BE has not been started yet :)